weasyl/test/web/test_restricted_tags.py

```diff
@@ -0,0 +1,206 @@
+    resp = app.get('/directorcontrol/globaltagrestrictions', headers={'Cookie': director_cookie})
+    assert "No community tags are currently restricted." in resp
+
+
+@pytest.mark.usefixtures('db')
+def test_ensure_non_directors_cannot_access_global_restricted_tags(monkeypatch):
+    user = db_utils.create_user()
+    user_cookie = db_utils.create_session(user)
+    resp = app.get('/directorcontrol/globaltagrestrictions', headers={'Cookie': user_cookie})
+    assert resp.html.find(id='error_content').p.string == errorcode.permission
+
+    monkeypatch.setattr(staff, 'DEVELOPERS', frozenset([user]))
+    resp = app.get('/directorcontrol/globaltagrestrictions', headers={'Cookie': user_cookie})
+    assert resp.html.find(id='error_content').p.string == errorcode.permission
+
+    monkeypatch.setattr(staff, 'MODS', frozenset([user]))
+    resp = app.get('/directorcontrol/globaltagrestrictions', headers={'Cookie': user_cookie})
+    assert resp.html.find(id='error_content').p.string == errorcode.permission
+
+    monkeypatch.setattr(staff, 'TECHNICAL', frozenset([user]))
+    resp = app.get('/directorcontrol/globaltagrestrictions', headers={'Cookie': user_cookie})
+    assert resp.html.find(id='error_content').p.string == errorcode.permission
+
+    monkeypatch.setattr(staff, 'ADMINS', frozenset([user]))
+    resp = app.get('/directorcontrol/globaltagrestrictions', headers={'Cookie': user_cookie})
+    assert resp.html.find(id='error_content').p.string == errorcode.permission
+
+
+@pytest.mark.usefixtures('db')
+def test_view_inuse_user_and_global_restricted_tags_page(restricted_tags):
+    _, user_cookie, _, director_cookie = restricted_tags
+
+    resp = app.get('/control/tagrestrictions', headers={'Cookie': user_cookie})
+    assert resp.html.find(id='tags-textarea').string == tag_user
+
+    resp = app.get('/directorcontrol/globaltagrestrictions', headers={'Cookie': director_cookie})
+    assert resp.html.find(id='tags-textarea').string == tag_global
+
+
+@pytest.mark.usefixtures('db')
+def test_verify_that_csrf_check_exists_on_user_and_global_restricted_tags(restricted_tags):
+    _, user_cookie, _, director_cookie = restricted_tags
+
+    # User and Global restricted tags POST routes require CSRF; verify this exists and functions
+    resp = app.post('/control/tagrestrictions', {'tags': tag_user}, headers={'Cookie': user_cookie})
+    assert resp.html.find(id='error_content').p.string == errorcode.token
+    resp = app.post('/directorcontrol/globaltagrestrictions', {'tags': tag_global}, headers={'Cookie': director_cookie})
+    assert resp.html.find(id='error_content').p.string == errorcode.token
+
+
+@pytest.mark.usefixtures('db', 'no_csrf')
+def test_adding_user_and_global_restricted_tags(restricted_tags):
+    user, user_cookie, director, director_cookie = restricted_tags
+
+    # Clear restricted tags
+    searchtag.edit_user_tag_restrictions(user, set())
+    searchtag.edit_user_tag_restrictions(director, set())
```

Comment:
Now that I look at this, these lines (L125/126) are useless in this context; I'll be editing these two lines out via ``git commit --amend`` when I get back to my computer that has a git CLI.